packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -174,9 +218,59 @@ class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
     }
   }
 
+  @override
+  void dispose() {
+    for (AnimationController animationController in selectionControllers) {
+      animationController.dispose();
+    }
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(SegmentedControl<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.groupValue != widget.groupValue) {
+      int index = 0;
+      for (T key in widget.children.keys) {
+        if (widget.groupValue == key) {
+          childTweens.insert(index, forwardBackgroundColorTween);
+          selectionControllers.elementAt(index).forward();
+        } else {
+          childTweens.insert(index, reverseBackgroundColorTween);
+          selectionControllers.elementAt(index).reverse();
+        }
+        index += 1;
+      }
+    }
+  }
+
+  Color getTextColor(int index, T currentKey) {
+    if (selectionControllers.elementAt(index).isAnimating) {
+      return textColorTween.evaluate(selectionControllers.elementAt(index));
+    } else if (widget.groupValue == currentKey) {
+      return CupertinoColors.white;
+    } else {
+      return CupertinoColors.activeBlue;
+    }
```

Comment:
minor nit: it's generally considered better style to not have an "else" after a "return", since the return means that you couldn't continue anyway.